test/functional/rpc_net.py

```diff
@@ -101,5 +104,33 @@ def _test_getpeerinfo(self):
         assert_equal(peer_info[0][0]['minfeefilter'], Decimal("0.00000500"))
         assert_equal(peer_info[1][0]['minfeefilter'], Decimal("0.00001000"))
 
+    def _test_getnodeaddresses(self):
+        self.nodes[0].add_p2p_connection(P2PInterface())
+
+        # send some addresses to the node via the p2p message addr
+        msg = msg_addr()
+        imported_addrs = []
+        for i in range(256):
+            a = "123.123.123.{}".format(i)
+            imported_addrs.append(a)
+            addr = CAddress()
+            addr.time = 100000000
+            addr.nServices = NODE_NETWORK | NODE_WITNESS
+            addr.ip = a
+            addr.port = 8333
+            msg.addrs.append(addr)
+        self.nodes[0].p2p.send_and_ping(msg)
+
+        # obtain addresses via rpc call and check they were ones sent in before
+        NODE_ADDRESSES_REQUEST_COUNT = 10
+        node_addresses = self.nodes[0].getnodeaddresses(NODE_ADDRESSES_REQUEST_COUNT)
```

Comment:
Use named arguments instead of long and verbose variables? Then ditch `NODE_ADDRESSES_REQUEST_COUNT`.

Reply:
Could also test that the result is limited by the existing addressed, not by the requested count.

Reply:
Unless I'm missing something, that verbose variable is still needed for the assert_equal check in the line after. It seems a better way than having magic numbers. I've reduced the length of the variable name.